addons/website_rating/models/mail_message.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class MailMessage(models.Model):
+    _inherit = 'mail.message'
+
+    rating_value = fields.Float("Rating Value", compute='_compute_rating_value', search='_search_rating_value')
+
+    @api.multi
+    def _compute_rating_value(self):
+        ratings = self.env['rating.rating'].search([('message_id', 'in', self.ids), ('consumed', '=', True)], order='create_date DESC')
+        mapping = dict((r.message_id.id, r.rating) for r in ratings)
+        for message in self:
+            message.rating_value = mapping.get(message.id, 0.0)
+
+    def _search_rating_value(self, operator, value):
+        self.env.cr.execute("""
+            SELECT rating, message_id
+            FROM rating_rating
+            WHERE rating = %s AND consumed='t' AND message_id IS NOT NULL
+            ORDER BY create_date DESC""", (value,)
+        )
+        message_ids = [m['message_id'] for m in self.env.cr.dictfetchall()]
+        return [('id', 'in', message_ids)]
+
+    @api.multi
+    def website_message_format(self):
+        result = super(MailMessage, self).website_message_format()
+        # inlude rating value in data if requested
+        if self._context.get('rating_include'):
+            message_data = self.read(['rating_value'])
+            message_tree = dict((m['id'], m['rating_value']) for m in message_data)
```

Comment:
Use dictionary comprehension, and is there reason to use read instead of regular browsing?